lnst/Recipes/ENRT/ConfigMixins/LongLivedConnectionsMixin.py

```diff
@@ -0,0 +1,212 @@
+    def generate_sub_configurations(self, config):
+        for parent_config in super().generate_sub_configurations(config):
+            parent_config.long_lived_connections = []
+
+            for ip_version in self.params.ip_versions:
+                host1, host2 = self.matched.host1, self.matched.host2
+                filtered_ips = zip(
+                    config.long_lived_ips[host1][ip_version],
+                    config.long_lived_ips[host2][ip_version],
+                )
+
+                for i, endpoint_pair in enumerate(filtered_ips):
+                    generator_ip, receiver_ip = endpoint_pair
+
+                    connections_count = self.calculate_client_connections(i)
+
+                    server_job = self._prepare_server(
+                        host2.eth0, receiver_ip, connections_count
+                    )
+                    client_job = self._prepare_client(
+                        host1.eth0, receiver_ip, generator_ip, connections_count
+                    )
+
+                    parent_config.long_lived_connections.append(
+                        (client_job, server_job)
+                    )
+
+            yield parent_config
+
+    def calculate_client_connections(self, client_id):
+        client_id += 1  # 0-based index
+        if client_id < self.servers_count:
+            return self.params.long_lived_conns_per_ip
+
+        opened_connections = (
+            self.servers_count - 1
+        ) * self.params.long_lived_conns_per_ip
+
+        # remaining connections are handled by last client
+        return self.params.long_lived_conns - opened_connections
+
+    def _prepare_server(self, receiver_nic, receiver_ip, conns_count):
+        server = LongLivedServer(
+            server_ip=receiver_ip,
+            server_port=self.params.long_lived_conns_port,
+            connections_count=conns_count,
+        )
+
+        job = receiver_nic.netns.prepare_job(server)
+
+        return job
+
+    def _prepare_client(self, generator_nic, receiver_ip, generator_ip, conns_count):
+        client = LongLivedClient(
+            server_ip=receiver_ip,
+            server_port=self.params.long_lived_conns_port,
+            client_ip=generator_ip,
+            connections_count=conns_count,
+        )
+
+        job = generator_nic.netns.prepare_job(client)
+
+        return job
+
+    def generate_sub_configuration_description(self, config):
+        desc = super().generate_sub_configuration_description(config)
+
+        for client_job, server_job in config.long_lived_connections:
+            desc.append(
+                f"Long-lived connection between {client_job.what} and {server_job.what}"
+            )
+
+        return desc
+
+    def apply_sub_configuration(self, config):
+        super().apply_sub_configuration(config)
+
+        for client_job, server_job in config.long_lived_connections:
+            server_job.start(bg=True)
+            time.sleep(2)  # just to be sure the server is up
```

Comment:
depending on the number of client-server pairs it may be more sensible to split this into:
```
for _, server_job in config.long_lived_connections:
    server_job.start(bg=True)
time.sleep(2) # wait for all servers to start
for client_job, _ in config.long_lived_connections:
    client_job.start(bg=True)
```

this could save some extra sleeps.